server/src/main/java/org/elasticsearch/common/blobstore/support/FilterBlobContainer.java

```diff
@@ -40,18 +41,18 @@ public BlobPath path() {
     }
 
     @Override
-    public boolean blobExists(String blobName) throws IOException {
-        return delegate.blobExists(blobName);
+    public boolean blobExists(OperationPurpose purpose, String blobName) throws IOException {
+        return delegate.blobExists(OperationPurpose.SNAPSHOT, blobName);
```

Comment:
Throughout `FilterBlobContainer` we should pass the `purpose` argument rather than hard-coding it to `SNAPSHOT`.

Reply:
I missed this file. Thanks for catching it. Your comment prompted me to do another sweep to check where the default value of `SNAPSHOT` is used. There are two scenarios where the default value is used:
1. Oversight like this one where this argument should be passed through from the caller method, also found in files such as FsBlobContainer etc. These are now fixed.
2. The actual call site or its approximate. An example is BlobStoreRepository or ChecksumBlobStoreFormat. These are the classes where we will leverage change of this PR to pass in the actual operation purpose. The equivalent on the stateless side is the ObjectStoreService. I think it makes sense to have them as follow-ups.